bccsp/schemes/dlog/crypto/signature.go

```diff
@@ -841,6 +869,28 @@ func (sig *Signature) Ver(
 		if !Nym_eid.Equals(EidNym) {
 			return errors.Errorf("signature invalid: nym eid validation failed")
 		}
+
+		if !meta.NymEIDAuditData.Nym.Equals(EidNym) {
+			return errors.Errorf("signature invalid: nym eid validation failed")
+		}
+	}
+
+	if meta != nil && len(meta.NymEID) != 0 {
```

Comment:
Is this check supposed to be here? I think it should only be executed if `verifyEIDNym` is true, otherwise we might fork any users of this library: consider this scenario:
* a consensus-aware application upgrades to the latest version of the library with this patch
* the library still calls `Ver` to verify signatures with the default settings (`verifyEIDNym` is false)
* an adversary manages to have the consensus-aware application call `Ver` while supplying a non-nil `meta`
* at this point if the network of consensus-aware applications don't all run the exact same version of the library (e.g. because they're going through a rolling upgrade), one node running the old version of the library will ignore meta while another node running a new version of the library will look at meta and might err out and fork the network

Wdyt?

Reply:
agreed, moved under verifyEIDNym